praw/models/trophy.py

```diff
@@ -45,6 +46,16 @@ def __init__(self, reddit: Reddit, _data: Dict[str, Any]):
         assert isinstance(_data, dict) and "name" in _data
         super().__init__(reddit, _data=_data)
 
+    def __eq__(self, other: _Trophy) -> bool:
+        """Check if two Trophies are equal."""
+        if isinstance(other, self.__class__):
```

Comment:
Your type annotation indicates `other` must be a trophy, but your check about whether `other` is an instance of `self.__class__` indicates that `other` could be anything. Which is it? I suspect the type annotation is wrong — after all, I could try to compare a trophy and something else with `trophy == other_thing`, and that's a valid operation which would evaluate to `False`.

Reply:
I will update the type references, but yes, while it can be `Any`, I wanted to show that it's really only ever going to return True with another Trophy.